certauth.go

```diff
@@ -70,14 +86,21 @@ func (a *Auth) Handler(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// Let secure process the request. If it returns an error,
 		// that indicates the request should not continue.
-		err := a.Process(w, r)
-
-		// If there was an error, do not continue.
-		if err != nil {
+		if err := a.Process(w, r); err != nil {
+			// if process returned an error request should not continue
 			return
 		}
 
-		h.ServeHTTP(w, r)
+		ctx := r.Context()
+		if len(a.opt.AllowedOUs) > 0 {
+			ctx = context.WithValue(r.Context(), HasAuthorizedOU, r.TLS.VerifiedChains[0][0].Subject.OrganizationalUnit)
+		}
+
+		if len(a.opt.AllowedCNs) > 0 {
+			ctx = context.WithValue(r.Context(), HasAuthorizedCN, r.TLS.VerifiedChains[0][0].Subject.CommonName)
```

Comment:
Wouldn't we prefer to pass `WithValue(` the local var `ctx` instead of grabbing `r.Context()`?

Doing so would allow the context to contain both `HasAuthorizedOU` and `HasAuthorizedCN`.